src/hevm/src/EVM.hs

```diff
@@ -1963,15 +1966,13 @@ create self this xGas xValue xs newAddr initCode = do
 
         zoom (env . contracts) $ do
           oldAcc <- use (at newAddr)
-          let oldBal = case oldAcc of
-                Nothing -> 0
-                Just c  -> view balance c
-          assign (at newAddr) (Just newContract)
-          assign (ix newAddr . balance) (oldBal + xValue)
-          assign (ix newAddr . nonce) 1
-          modifying (ix self . balance) (flip (-) xValue)
+          let oldBal = maybe 0 (view balance) oldAcc
+
+          assign (at newAddr) (Just (newContract & balance .~ oldBal))
```

Comment:
whose `balance` is being written to here?

Reply:
`newContract`